bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/servlets/TriggerDeckDynamoServlet.java

```diff
@@ -0,0 +1,121 @@
+package com.adobe.acs.commons.indesign.dynamicdeckdynamo.servlets;
+
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.exception.DynamicDeckDynamoException;
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.models.DynamicDeckInitiatorPageModel;
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.services.DynamicDeckService;
+import com.adobe.granite.rest.Constants;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.ServletResolverConstants;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.servlets.post.JSONResponse;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Arrays;
+import java.util.List;
+
+@Component(
+        service = Servlet.class,
+        property = {
+                ServletResolverConstants.SLING_SERVLET_RESOURCE_TYPES + "=" + ServletResolverConstants.DEFAULT_RESOURCE_TYPE,
+                ServletResolverConstants.SLING_SERVLET_METHODS + "=" + HttpConstants.METHOD_GET,
+                ServletResolverConstants.SLING_SERVLET_EXTENSIONS + "=" + "json",
+                ServletResolverConstants.SLING_SERVLET_SELECTORS + "=" + TriggerDeckDynamoServlet.SELECTOR_TRIGGER_DECK_DYNAMO})
+public class TriggerDeckDynamoServlet extends SlingAllMethodsServlet {
+
+    public static final String SELECTOR_TRIGGER_DECK_DYNAMO = "triggerDeckDynamo";
+    private static final Logger LOGGER = LoggerFactory.getLogger(TriggerDeckDynamoServlet.class);
+    private static final long serialVersionUID = 2151877091112583303L;
+    private static final String MESSAGE = "message";
+
+    @Reference
+    private transient DynamicDeckService dynamicDeckService;
+
+    @Override
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
```

Comment:
Should this be a doPost? since it materially changes underlying state of the application?

Also, if not - then it should be `SlingSafeMethodsServlet`